website/project/views/comment.py

```diff
@@ -13,106 +14,169 @@
+        users.keys(),
+        key=lambda item: get_recency(item),
+        reverse=True,
+    )
+
     return {
-        'discussion': [
-            {
-                'id': privacy_info_handle(user._id, anonymous),
-                'url': privacy_info_handle(user.url, anonymous),
-                'fullname': privacy_info_handle(user.fullname, anonymous, name=True),
-                'isContributor': node.is_contributor(user),
-                'gravatarUrl': privacy_info_handle(
-                    gravatar(
-                        user, use_ssl=True, size=settings.GRAVATAR_SIZE_DISCUSSION,
-                    ),
-                    anonymous
-                ),
-
-            }
-            for user in sorted_users
+        'discussionByFrequency': [
+            serialize_discussion(node, user, len(users[user]), anonymous)
+            for user in sorted_users_frequency
+        ],
+        'discussionByRecency': [
+            serialize_discussion(node, user, len(users[user]), anonymous)
+            for user in sorted_users_recency
         ]
     }
 
 
+def serialize_discussion(node, user, num, anonymous=False):
+    return {
+        'id': privacy_info_handle(user._id, anonymous),
+        'url': privacy_info_handle(user.url, anonymous),
+        'fullname': privacy_info_handle(user.fullname, anonymous, name=True),
+        'isContributor': node.is_contributor(user),
+        'gravatarUrl': privacy_info_handle(
+            gravatar(
+                user, use_ssl=True, size=settings.GRAVATAR_SIZE_DISCUSSION,
+            ),
+            anonymous
+        ),
+        'numOfComments': num
+    }
+
+
 def serialize_comment(comment, auth, anonymous=False):
+    node = comment.node
+    if hasattr(comment.root_target, 'page_name'):  # Wiki
+        # In case the wiki name is changed
+        root_id = comment.root_target.page_name
+        title = comment.root_target.page_name
+    elif isinstance(comment.root_target, GuidFile):  # File
+        root_id = comment.root_target._id
+        title = comment.root_target.waterbutler_path
+    else:  # Node or comment
+        root_id = comment.root_target._id
+        title = ''
     return {
         'id': comment._id,
-        'author': {
-            'id': privacy_info_handle(comment.user._id, anonymous),
-            'url': privacy_info_handle(comment.user.url, anonymous),
-            'name': privacy_info_handle(
-                comment.user.fullname, anonymous, name=True
-            ),
-            'gravatarUrl': privacy_info_handle(
-                gravatar(
-                    comment.user, use_ssl=True,
-                    size=settings.GRAVATAR_SIZE_DISCUSSION
-                ),
-                anonymous
-            ),
-        },
+        'author': serialize_discussion(node, comment.user, 1, anonymous),
         'dateCreated': comment.date_created.isoformat(),
         'dateModified': comment.date_modified.isoformat(),
+        'page': comment.page or Comment.OVERVIEW,
```

Comment:
Why default to `Comment.OVERVIEW`? Shouldn't `comment` always have a `page` attribute at this point?